controllers/api_v1/keyword_test.go

```diff
@@ -0,0 +1,202 @@
+}
+
+func (s *KeywordAPIControllerDbTestSuite) TestUploadKeywordAPIWithoutAuthorizationHeader() {
+	resp := testHttp.PerformRequest(s.engine, "POST", "/api/v1/keywords", nil, nil)
+	respBodyData, _ := ioutil.ReadAll(resp.Body)
+	var parsedRespBody map[string][]api_helper.ErrorResponseObject
+	_ = json.Unmarshal(respBodyData, &parsedRespBody)
+
+	assert.Equal(s.T(), http.StatusUnauthorized, resp.Code)
+	assert.Equal(s.T(), errors.ErrInvalidAccessToken.Error(), parsedRespBody["errors"][0].Detail)
+}
+
+func (s *KeywordAPIControllerDbTestSuite) TestUploadKeywordAPIWithInvalidAccessToken() {
+	headers := http.Header{}
+	headers.Set("Authorization", "invalid_token")
+
+	resp := testHttp.PerformRequest(s.engine, "POST", "/api/v1/keywords", headers, nil)
+	respBodyData, _ := ioutil.ReadAll(resp.Body)
+	var parsedRespBody map[string][]api_helper.ErrorResponseObject
+	_ = json.Unmarshal(respBodyData, &parsedRespBody)
+
+	assert.Equal(s.T(), http.StatusUnauthorized, resp.Code)
+	assert.Equal(s.T(), errors.ErrInvalidAccessToken.Error(), parsedRespBody["errors"][0].Detail)
+}
+
+func (s *KeywordAPIControllerDbTestSuite) TestUploadKeywordAPIWithExpiredAccessToken() {
+	db.GetDB().Exec("DELETE FROM oauth2_tokens")
+
+	tokenItem := &oauth_test.TokenStoreItem{
+		CreatedAt: time.Now(),
+		ExpiresAt: time.Now(),
+		Code:      "test-code",
+		Access:    "test-access",
+		Refresh:   "test-refresh",
+	}
+
+	data, _ := json.Marshal(&oauth_test.TokenData{
+		AccessExpiresIn: 1,
+		Access:          tokenItem.Access,
+	})
+	tokenItem.Data = data
+
+	db.GetDB().Exec("INSERT INTO oauth2_tokens(created_at, expires_at, code, access, refresh, data) VALUES(?, ?, ?, ?, ?, ?)",
+		tokenItem.CreatedAt,
+		tokenItem.ExpiresAt,
+		tokenItem.Code,
+		tokenItem.Access,
+		tokenItem.Refresh,
+		tokenItem.Data,
+	)
+
+	headers := http.Header{}
+	headers.Set("Authorization", "Bearer test-access")
+
+	resp := testHttp.PerformRequest(s.engine, "POST", "/api/v1/keywords", headers, nil)
+	respBodyData, _ := ioutil.ReadAll(resp.Body)
+	var parsedRespBody map[string][]api_helper.ErrorResponseObject
+	_ = json.Unmarshal(respBodyData, &parsedRespBody)
+
+	assert.Equal(s.T(), http.StatusUnauthorized, resp.Code)
+	assert.Equal(s.T(), errors.ErrExpiredAccessToken.Error(), parsedRespBody["errors"][0].Detail)
+}
+
+func (s *KeywordAPIControllerDbTestSuite) TestUploadKeywordAPIWithAccessTokenButNoFileFormInTheRequest() {
+	headers := http.Header{}
+	headers.Set("Authorization", "Bearer test-access")
+
+	resp := testHttp.PerformRequest(s.engine, "POST", "/api/v1/keywords", headers, nil)
+	respBodyData, _ := ioutil.ReadAll(resp.Body)
+	var parsedRespBody map[string][]api_helper.ErrorResponseObject
+	_ = json.Unmarshal(respBodyData, &parsedRespBody)
+
+	assert.Equal(s.T(), http.StatusBadRequest, resp.Code)
+	assert.Equal(s.T(), "invalid file", parsedRespBody["errors"][0].Detail)
+}
+
+func (s *KeywordAPIControllerDbTestSuite) TestUploadKeywordAPIWithAccessTokenButBlankPayload() {
```

Comment:
should we add a test for `non-csv-type` @gutakk

Reply:
As every controllers call business logic in the service, I already have the test for validate the file type at the service level [here](https://github.com/gutakk/go-google-scraper/blob/development/services/keyword_service/keyword_test.go#L186-L190)

Do you think do we still need this test case in the controller?

Reply:
Ah saw it @gutakk As we have the test at the service level then it's ok @gutakk